graphql/atomic-integration_test.go

```diff
@@ -0,0 +1,129 @@
+	"github.com/evergreen-ci/evergreen/service"
+	"github.com/evergreen-ci/evergreen/testutil"
+	"github.com/mongodb/grip"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+)
+
+type atomicGraphQLSuite struct {
+	url       string
+	apiUser   string
+	apiKey    string
+	directory string
+
+	suite.Suite
+}
+
+func TestAtomicGraphQLSuite(t *testing.T) {
+	testDirectories, err := ioutil.ReadDir("tests")
+	require.NoError(t, err)
+	for _, dir := range testDirectories {
+		suite.Run(t, &atomicGraphQLSuite{directory: dir.Name()})
+	}
+}
+
+func (s *atomicGraphQLSuite) SetupSuite() {
+	const apiKey = "testapikey"
+	const apiUser = "testuser"
+
+	server, err := service.CreateTestServer(testutil.TestConfig(), nil, true)
+	s.Require().NoError(err)
+	env := evergreen.GetEnvironment()
+	ctx := context.Background()
+	s.Require().NoError(env.DB().Drop(ctx))
+	testUser := user.DBUser{
+		Id:          apiUser,
+		APIKey:      apiKey,
+		Settings:    user.UserSettings{Timezone: "America/New_York"},
+		SystemRoles: []string{"unrestrictedTaskAccess"},
+	}
+	s.Require().NoError(testUser.Insert())
+	s.url = server.URL
+	s.apiKey = apiKey
+	s.apiUser = apiUser
+}
+
+type testsCases struct {
+	Tests []test `json:"tests"`
+}
+
+type data map[string]json.RawMessage
+
+func (s *atomicGraphQLSuite) TestQueries() {
+	dataFile, err := ioutil.ReadFile(filepath.Join("tests", s.directory, "data.json"))
+	s.Require().NoError(err)
+
+	resultsFile, err := ioutil.ReadFile(filepath.Join("tests", s.directory, "results.json"))
+	s.Require().NoError(err)
+
+	var testData data
+	err = json.Unmarshal(dataFile, &testData)
+	s.Require().NoError(err)
+
+	var tests testsCases
+	err = json.Unmarshal(resultsFile, &tests)
+	s.Require().NoError(err)
+
+	s.Require().NoError(testData.SetupData(*evergreen.GetEnvironment().DB()))
+
+	for _, testCase := range tests.Tests {
+		singleTest := func(t *testing.T) {
+			f, err := ioutil.ReadFile(filepath.Join("tests", s.directory, "queries", testCase.QueryFile))
+			require.NoError(t, err)
+			jsonQuery := fmt.Sprintf(`{"operationName":null,"variables":{},"query":"%s"}`, escapeGQLQuery(string(f)))
+			body := bytes.NewBuffer([]byte(jsonQuery))
+			client := http.Client{}
+			r, err := http.NewRequest(http.MethodPost, fmt.Sprintf("%s/graphql/query", s.url), body)
+			s.Require().NoError(err)
+			r.Header.Add(evergreen.APIKeyHeader, s.apiKey)
+			r.Header.Add(evergreen.APIUserHeader, s.apiUser)
+			r.Header.Add("content-type", "application/json")
+			resp, err := client.Do(r)
+			require.NoError(t, err)
+			b, err := ioutil.ReadAll(resp.Body)
+			require.NoError(t, err)
+			assert.JSONEq(t, string(testCase.Result), string(b), fmt.Sprintf("expected %s but got %s", string(testCase.Result), string(b)))
+		}
+
+		s.T().Run(testCase.QueryFile, singleTest)
+	}
+}
+
+func (d *data) SetupData(db mongo.Database) error {
```

Comment:
I think we should drop whatever data was added in the previous test before adding data from the current test

Reply:
as you can see in method `TestAtomicGraphQLSuite`, `SetupData` is called for every directory in tests. i believe this will automatically create a new slate for each test suite. i just tested it by adding a new directory, added no new data in the new test directory, and then tried to query data from a previous test setup. it failed, which leads me to believe data for each test suite is isloated.